Targets/QJS/Patches/Fuzzilli-instrumentation-for-QJS.patch

```diff
@@ -131,198 +144,192 @@ index 7e4839f..7dbf2b6 100644
 -            if (eval_file(ctx, filename, module))
 -                goto fail;
-+          fprintf(stderr, "Unknown action: %u\n", action);
-+          _exit(-1);
-+        }
-+        char *script_src = (char *)(js_malloc(ctx, script_size+1));
-+
-+        char *ptr = script_src;
-+        size_t remaining = script_size;
-+        while (remaining > 0) {
-+          ssize_t rv = read(REPRL_DRFD, ptr, remaining);
-+          if (rv <= 0) {
-+            fprintf(stderr, "Failed to load script\n");
-+            _exit(-1);
-+          }
-+          remaining -= rv;
-+          ptr += rv;
-         }
+-        }
 -        if (interactive) {
 -            js_std_eval_binary(ctx, qjsc_repl, qjsc_repl_size, 0);
-+        script_src[script_size] = '\0';
-+        if (0 == (result = eval_buf(ctx, script_src, script_size, "reprl", 0))) {
-+          js_std_loop(ctx);
-+        } else {
-+          fprintf(stderr, "Failed to eval_buf reprl\n");
++            if (expr) {
++                if (eval_buf(ctx, expr, strlen(expr), "<cmdline>", 0))
++                    goto fail;
++            } else
++            if (optind >= argc) {
++                /* interactive mode */
++                interactive = 1;
++            } else {
++                const char *filename;
++                filename = argv[optind];
++                if (eval_file(ctx, filename, module))
++                    goto fail;
++            }
++            // if (interactive) {
++            //     js_std_eval_binary(ctx, qjsc_repl, qjsc_repl_size, 0);
++            // }
++            js_std_loop(ctx);
          }
 -        js_std_loop(ctx);
-+        js_free(ctx, script_src);
-+        fflush(stdout);
-+        fflush(stderr);
-+        // Send return code to parent and reset edge counters.
-+        status = (result & 0xff) << 8;
-+        CHECK(write(REPRL_CWFD, &status, 4) == 4);
-+        __sanitizer_cov_reset_edgeguards();
-+      } while (reprl_mode);
-+    } else {
-+      if (expr) {
-+        if (eval_buf(ctx, expr, strlen(expr), "<cmdline>", 0)) goto fail;
-+      } else if (optind >= argc) {
-+        /* interactive mode */
-+        interactive = 1;
-+        // Google Change
-+        // if (interactive) {
-+        //  js_std_eval_binary(ctx, qjsc_repl, qjsc_repl_size, 0);
-+        // }
-+        // End Google Change
-+      } else {
-+        const char *filename;
-+        filename = argv[optind];
-+        if (eval_file(ctx, filename, module)) goto fail;
-+      }
-+      js_std_loop(ctx);
      }
-+  }
      
      if (dump_memory) {
-         JSMemoryUsage stats;
 diff --git a/quickjs.c b/quickjs.c
-index 8fbb7a3..a9d1110 100644
+index 7916013..f84825a 100644
 --- a/quickjs.c
 +++ b/quickjs.c
-@@ -22,11 +22,13 @@
-  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
-  * THE SOFTWARE.
-  */
-+
- #include <stdlib.h>
- #include <stdio.h>
+@@ -27,6 +27,7 @@
  #include <stdarg.h>
  #include <inttypes.h>
  #include <string.h>
 +#undef NDEBUG
```

Comment:
I realize this was here before, but is this actually required? Shouldn't debug or no-debug configuration be configured elsewhere?

Reply:
Yeah I didn't think these did anything because NDEBUG isn't being defined anywhere else as far as I can tell, but I left them just in case.  I've removed this line in both the qjs.c and quickjs.c files in the patch.